tests/testthat/test-new-model.R

```diff
@@ -146,3 +146,41 @@ test_that("new_model() supports `.path` containing a period [BBR-NWMD-010]", {
   mod <- new_model(fs::path_ext_remove(temp_ctl), "path with period")
   expect_true(fs::file_exists(temp_yaml))
 })
+
+test_that("new_model accepts .ctl file extension when passed [BBR-CLM-005]", {
```

Comment:
As a suggest elsewhere, I think it's fine to just test one extension, but if you keep testing both:

 * these test IDs can be rolled up into each other (e.g., "new_model ignores file extension: .ctl [BBR-CLM-005]" and "new_model ignores file extension: .mod [BBR-CLM-005]")
 * these are sharing a lot of code and it looks like all the assertions, so it'd be nice to avoid the duplication.  (See `BBR-SUM-003` for an example.)